chia/wallet/wallet_node.py

```diff
@@ -1199,9 +1198,14 @@ async def sync_from_untrusted_close_to_peak(self, new_peak_hb: HeaderBlock, peer
             # For every block, we need to apply the cache from race_cache
             for potential_height in range(backtrack_fork_height + 1, new_peak_hb.height + 1):
                 header_hash = self.wallet_state_manager.blockchain.height_to_hash(uint32(potential_height))
-                if header_hash in self.race_cache:
-                    self.log.info(f"Receiving race state: {self.race_cache[header_hash]}")
-                    await self.add_states_from_peer(list(self.race_cache[header_hash]), peer)
+                race_cache = self.race_cache.get(header_hash)
+                if race_cache is not None:
+                    self.log.info(f"Receiving race cache: {race_cache}")
+                    if await self.add_states_from_peer(list(race_cache.coin_states), peer):
+                        self.race_cache.pop(header_hash)
+
+            # Clear old entries that are no longer relevant
+            self.cleanup_race_cache(min_height=new_peak_hb.height - 100)
```

Comment:
do do we need the race cache for reorgs? I would have expected the height to not subtract 100 here

Reply:
No we don't need it for reorgs, the reorgs come in as new updates anyway and the 100 is just inherited from what it was before. I moved the cache cleanup out of ` add_state_to_race_cache`. See few lines above.